nautobot/utilities/forms/forms.py

```diff
@@ -233,3 +244,107 @@ def __init__(self, table, *args, **kwargs):
     @property
     def table_name(self):
         return self.table.__class__.__name__
+
+
+class DynamicFilterForm(BootstrapMixin, forms.Form):
+    """
+    Form for dynamically inputting filter values for an object list.
+    """
+
+    lookup_field = forms.ChoiceField(
+        choices=[],
+        required=False,
+        label="Field",
+    )
+    lookup_type = forms.ChoiceField(
+        choices=[],
+        required=False,
+    )
+    lookup_value = forms.CharField(
+        required=False,
+        label="Value",
+    )
+
+    def __init__(self, *args, filterset=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        from nautobot.utilities.forms import add_blank_choice  # Avoid circular import
+
+        # cls.model is set at `dynamic_formset_factory()`
+        self.filterset = filterset or getattr(self, "filterset", None)
+
+        # Raise exception if `cls.filterset` not set and `filterset` not passed
+        if self.filterset is None:
+            raise AttributeError("'DynamicFilterForm' object requires `filterset` attribute")
+
+        model = self.filterset._meta.model
+
+        if self.filterset is not None:
+            filterset = self.filterset()
+            self.filterset_filters = filterset.filters
+            contenttype = model._meta.app_label + "." + model._meta.model_name
+
+            # Configure fields: Add css class and set choices for lookup_field
+            self.fields["lookup_field"].choices = add_blank_choice(self._get_lookup_field_choices())
+            self.fields["lookup_field"].widget.attrs["class"] = "nautobot-select2-static lookup_field-select"
+
+            # Update lookup_type and lookup_value fields to match expected field types derived from data
+            # e.g status expects a ChoiceField with APISelectMultiple widget, while name expects a CharField etc.
+            if "data" in kwargs and "prefix" in kwargs:
+                data = kwargs["data"]
+                prefix = kwargs["prefix"]
+                lookup_type = data.get(prefix + "-lookup_type")
+                lookup_value = data.getlist(prefix + "-lookup_value")
+
+                if lookup_type and lookup_value and lookup_type in self.filterset_filters:
```

Comment:
if `lookup_type` is not in `self.filterset_filters`, is that something we should log a warning or error about?

Reply:
In the UI, a warning would have been issued for this.

Reply:
Hmm. How would the UI trigger this case? If it's not something that should be possible under normal usage, I feel like a server-side log message would be appropriate.

Reply:
Before using the `DynamicFilterSet`, `filterset = self.filterset(filter params, self.queryset)` in the `ObjectListView` handled this invalid field.

However, I believe it's fantastic to also log a warning.